src/commands/infra/generate.test.ts

```diff
@@ -83,10 +84,8 @@ describe("Validate template path from a definition.json", () => {
 describe("Validate spk.tfvars file", () => {
   test("Validating that a spk.tfvars is generated and has appropriate format", async () => {
     const mockProjectPath = "src/commands/infra/mocks";
-    const generateTfvars = await generateSpkTfvars(
-      mockProjectPath,
-      mockProjectPath
-    );
+    const definitionJSON = await readDefinitionJson(mockProjectPath);
```

Comment:
@yradsmikham this unit test appears to be doing file system operations. If we separated the filesystem input portion from the actual JSON objects in `writeSpkTfvars` it seems we could have unit tests that are not dependent on looking at the file system. This makes the tests more portable as well. 

We should consider decoupling the filesystem operations from the business logic in `writeSpkTfvars` when making further PRs

Reply:
The challenge is decoupling `fs` operations in `writeSpkTfvars` because this is the one method in the command that will create and write to the `spk.tfvars`. So, we _can_ decouple_ fs operations into another method, and call the method within `writeSpkTfcars`, but it would only make the script more complex, and lead us to write another unit test to test additional methods.

Do you think this is more of an integration test? If so, we could move this over.

Reply:
@yradsmikham I would argue it makes the business logic easier to see and less complicated since your method is smaller. I can share an example offline with you about this when I get time. All I'm saying is for future PRs let's think about it. 

Technically this is an integration test. I'm saying we should avoid disk and network I/O in unit tests (for speed, portability, and designing our code to have clear dependencies). If we can work towards that great but I understand sometimes it may be difficult or non-obvious. 

One thing I realized from this is that we don't publish results of how long each "unit" test is taking. Once we have those result we can make a data driven decision on if this is worth the time.

Reply:
Second the removal of fs ops from unit tests.

Reply:
fixed in my latest commit!